plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1690,3 +1690,65 @@ def calculate_checksum(file_handle):
         :rtype:             string
         """
         return file_utils.calculate_checksum(file_handle)
+
+
+class ModulemdDefaults(FileContentUnit):
+    """
+    This model represents a Modulemd-defaults metadata document.
+
+    There could be only one modulemd-defaults per module name in one repo,
+    thus repo_id is a part of the unit_key.
+
+    :param name: Module name that the defaults are for
+    :type  name: mongoengine.StringField
+
+    :param repo_id: Id of a repository the modulemd-defaults belongs to
+    :type  repo_id: mongoengine.StringField
+
+    :param stream: Module stream that is the default for the module
+    :type  stream: mongoengine.StringField
+
+    :param profiles: Module profiles indexed by the stream name. This is a dictionary of stream
+                     names to a list of default profiles to be installed. This dictionary is
+                     encoded into BSON format because MongoDB doesn't allow keys to contain dots.
+                     When decoded back from BSON format it is a proper Python dict.
+    :type  profiles: mongoengine.StringField
+
+    :param checksum: Checksum of the modulemd-defaults document
+    :type  checksum: mongoengine.StringField
+    """
+
+    # Unit key fields
+    name = mongoengine.StringField(required=True)
+    repo_id = mongoengine.StringField(required=True)
+
+    stream = mongoengine.StringField()
+    profiles = mongoengine.StringField()
+
+    checksum = mongoengine.StringField()
```

Comment:
what is the reason of this this field on the model? i *think* we should entirely store this in the DB and not on the filesystem like in case of modulemd document. In addition, defaults are mutable therefore it is not a good idea to store them on the filesystem .Was this the purpose of checksum field?

Reply:
We can store it in the DB only at the moment.
My reasoning to store it on the FS as well:
 - sync/upload/publish will work in a similar way for both content types (modulemd-defaults, modulemd)
 - yaml file where both types are present will be generated in the same way
 - we don't support intents yet but it seems like they can be of a noticable size and maybe we won't need to store everything from intents in the DB.

I believe defaults are immutable in Pulp terms. The requested logic is ability to overwrite the existing modulemd-defaults, upload and sync should take care of that. Right now the only mutable type in rpm plugin is errata. Only parts of it can  be changed and there is a logic to merge its parts. I don't think we need that for defaults. If you think we may need that, I suggest to avoid that as much as we can.

Reply:
@goosemania i've re-read the requirements and deliverables, i agree.